elfo-network/src/discovery/mod.rs

```diff
@@ -561,3 +597,5 @@ fn unexpected_message_error(envelope: Envelope, expected: &[&str]) -> eyre::Repo
 async fn timeout<T>(duration: Duration, fut: impl Future<Output = Result<T>>) -> Result<T> {
     tokio::time::timeout(duration, fut).await?
 }
+
+mod diff;
```

Comment:
move after `use` statements (and, optionally, replace `diff::X` with more explicit `self::diff::X`)